test/Mono.Linker.Tests.Cases/Reflection/TypeBaseTypeUseViaReflection.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	public class TypeBaseTypeUseViaReflection
+	{
+		public static void Main ()
+		{
+			KnownType_Derived.Test ();
+		}
+
+		[Kept]
+		class KnownType_Base
+		{
+			[Kept]
+			private static void UsedViaReflection () { }
+
+			private static void Unused () { }
+		}
+
+		[Kept]
+		[KeptBaseType (typeof (KnownType_Base))]
+		class KnownType_Derived : KnownType_Base
+		{
+			[Kept]
+			public static void Test ()
+			{
+				typeof (KnownType_Derived).BaseType.GetMethod ("UsedViaReflection", BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
Nit: this test would also succeed if we implemented BaseType as "return this".

Maybe add one more test that e.g. looks for a constructor on the base type.

Reply:
This uses private reflection, so calling it on the derived type should not find the method on the base type.
I added the `GetConstructor` as well.

Reply:
🤦 I see! Thanks